src/sentry/integrations/github/integration.py

```diff
@@ -458,67 +492,80 @@ def get_app_url(self) -> str:
-        try:
-            integration = Integration.objects.get(
-                external_id=installation_id, status=ObjectStatus.ACTIVE
-            )
-        except Integration.DoesNotExist:
-            return error(request, self.active_organization)
-
-        # Check that the authenticated GitHub user is the same as who installed the app.
-        if (
-            pipeline.fetch_state("github_authenticated_user")
-            != integration.metadata["sender"]["login"]
-        ):
-            return error(
-                request,
-                self.active_organization,
-                error_short="Authenticated user is not the same as who installed the app",
-            )
+            integration_pending_deletion_exists = False
+            if self.active_organization:
+                # We want to wait until the scheduled deletions finish or else the
+                # post install to migrate repos do not work.
+                integration_pending_deletion_exists = OrganizationIntegration.objects.filter(
+                    integration__provider=GitHubIntegrationProvider.key,
+                    organization_id=self.active_organization.organization.id,
+                    status=ObjectStatus.PENDING_DELETION,
+                ).exists()
+
+            if integration_pending_deletion_exists:
+                lifecycle.record_failure(
+                    {"failure_reason": GitHubInstallationError.PENDING_DELETION}
+                )
+                return error(
+                    request,
+                    self.active_organization,
+                    error_short=GitHubInstallationError.PENDING_DELETION,
+                    error_long=ERR_INTEGRATION_PENDING_DELETION,
+                )
+
+            try:
+                # We want to limit GitHub integrations to 1 organization
+                installations_exist = OrganizationIntegration.objects.filter(
+                    integration=Integration.objects.get(external_id=installation_id)
+                ).exists()
+
+            except Integration.DoesNotExist:
+                return pipeline.next_step()
+
+            if installations_exist:
+                lifecycle.record_failure(
+                    {"failure_reason": GitHubInstallationError.INSTALLATION_EXISTS}
+                )
+                return error(
+                    request,
+                    self.active_organization,
+                    error_short=GitHubInstallationError.INSTALLATION_EXISTS,
+                    error_long=ERR_INTEGRATION_EXISTS_ON_ANOTHER_ORG,
+                )
+
+            # OrganizationIntegration does not exist, but Integration does exist.
+            try:
+                integration = Integration.objects.get(
+                    external_id=installation_id, status=ObjectStatus.ACTIVE
+                )
+            except Integration.DoesNotExist:
+                lifecycle.record_failure({"failure_reason": "Integration does not exist"})
```

Comment:
Some of these errors are in the enum and some are not. Just commenting for consistency